src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -148,3 +148,22 @@ def test_query_freq_list_capabilities(session):
 
 def test_read_current_temperature(session):
     assert session.read_current_temperature() > 25.0
+
+
+def test_allocate_waveform(session):
+    session.allocate_waveform(1024)
```

Comment:
Verify this returns a valid waveform handle. You may need to do something to it afterwards, maybe an assert.

Reply:
Easier way to validate the changes will be writing more data than allocated.